doc/pset.mediawiki

```diff
@@ -234,6 +309,62 @@ determine which outputs are change outputs and verify that the change is returni
 ** Value: Any value data as defined by the proprietary type user.
 *** <tt><data></tt>
 
+The currently defined elements per-output proprietary types are as follows:
+
+* Type: Value <tt>PSET_OUT_VALUE = 0x00</tt>
+** Key: None. The key must only contain the 1 byte type.
+*** <tt>{0x00}</tt>
+** Value: The explicit little endian 64-bit integer value for this output.
+*** <tt>{value}</tt>
+
+* Type: Value Commitment <tt>PSET_OUT_VALUE_COMMITMENT = 0x01</tt>
+** Key: None. The key must only contain the 1 byte type.
+*** <tt>{0x01}</tt>
+** Value: The 33 byte Value Commitment for this output. Adding this field requires the removal of <tt>PSET_OUT_VALUE</tt>.
+*** <tt>{value commitment}</tt>
+
+* Type: Asset Tag <tt>PSET_OUT_ASSET = 0x02</tt>
+** Key: None. The key must only contain the 1 byte type.
+*** <tt>{0x02}</tt>
+** Value: The explicit 32 byte asset tag for this output.
+*** <tt>{asset}</tt>
+
+* Type: Asset Commitment <tt>PSET_OUT_ASSET_COMMITMENT = 0x03</tt>
+** Key: None. The key must only contain the 1 byte type.
+*** <tt>{0x03}</tt>
+** Value: The 33 byte Asset Commitment for this output. Adding this field requries the removal of <tt>PSET_OUT_ASSET</tt>.
+*** <tt>{asset commitment}</tt>
+
+* Type: Value Rangeproof <tt>PSET_OUT_VALUE_RANGEPROOF = 0x04</tt>
+** Key: None. The key must only contain the 1 byte type.
+*** <tt>{0x04}</tt>
+** Value: The rangeproof for the value of this output.
+*** <tt>{rangeproof}</tt>
+
+* Type: Asset Surjection Proof <tt>PSET_OUT_ASSET_SURJ_PROOF = 0x05</tt>
```

Comment:
if `RANGEPROOF` is spelled without underscore, it would be more consistent to spell `SURJPROOF` or `SURJECTIONPROOF` also without underscore. Or use `RANGE_PROOF` instead.

Reply:
I personally prefer `PSET_OUT_VALUE_RANGE_PROOF` and `PSET_OUT_ASSET_SURJECTION_PROOF`

Reply:
Although in the Elements code they are referred to as `rangeproof` and `surjectionproof`, without underscores.

Reply:
Also 'value' and 'asset' prefixes can be implicit in these names

Reply:
In the various papers on CT and CA, `rangeproof` as one word is the typical usage while `surjection proof` is 2 words. So following that convention, it would be `RANGEPROOF` and `SURJECTION_PROOF`.